### PR TITLE
Updating dockerfile and bumping dependencies

### PR DESCRIPTION
**Pull request by jeffgrunewald**
Breaking up the dockerfile flow to cache compiling dependencies so they only build once per revision of the rebar.config/rebar.lock file.

Also bumping dependencies including the latest release of blockchain-core with the latest snap.

### CODE REVIEW

File: config/sys.config [hunk 1]
```diff
@@ -33,10 +33,10 @@
         {block_sync_batch_limit, 100},
         {honor_quick_sync, true},
         {quick_sync_mode, blessed_snapshot},
-        {blessed_snapshot_block_height, 1108081},
+        {blessed_snapshot_block_height, 1130401},
         {blessed_snapshot_block_hash,
```
Comment (madninja) [inline config/sys.config:37]: there's a newer snapshot on s3 I believe? latest-snap.json shows that's the last one there

Comment (jeffgrunewald) [reply config/sys.config:37]: that's the latest available from latest-snap.json but i can update to one of the later ones if you can get me the hash/height

File: config/sys.config [hunk 2]
```diff
@@ -33,10 +33,10 @@
         {block_sync_batch_limit, 100},
         {honor_quick_sync, true},
         {quick_sync_mode, blessed_snapshot},
-        {blessed_snapshot_block_height, 1108081},
+        {blessed_snapshot_block_height, 1130401},
```
Comment (abhay) [inline config/sys.config]:
```suggestion
        {blessed_snapshot_block_height, 1139761},
```

File: config/sys.config [hunk 3]
```diff
@@ -33,10 +33,10 @@
...
         {blessed_snapshot_block_hash,
-            <<34, 134, 217, 35, 67, 96, 56, 194, 43, 8, 228, 103, 224, 24, 43, 104, 239, 52, 130,
-                6, 0, 169, 117, 78, 253, 85, 223, 165, 246, 75, 119, 181>>},
+            <<40,63,233,163,41,184,142,24,163,61,106,49,71,66,132,199,
+              141,169,106,160,26,125,124,217,36,45,89,222,50,106,41,251>>},
```
Comment (abhay) [inline config/sys.config]:
```suggestion
            <<146,107,41,218,41,184,179,87,36,238,57,89,37,225,247,176,220,148,143,
              243,201,73,107,134,10,222,98,162,35,85,217,67>>},
```

File: rebar.config [hunk 4]
```diff
@@ -29,10 +29,10 @@
     pbkdf2,
     {base64url, "1.0.1"},
     {observer_cli, "1.5.0"},
-    {elli, "3.2.0"},
+    {elli, "3.3.0"},
     {jsonrpc2, {git, "https://github.com/zuiderkwast/jsonrpc2-erlang.git", {branch, "master"}}},
     {clique, {git, "https://github.com/helium/clique.git", {branch, "develop"}}},
-    {blockchain, {git, "https://github.com/helium/blockchain-core.git", {tag, "2021.12.06.0"}}}
+    {blockchain, {git, "https://github.com/helium/blockchain-core.git", {tag, "2021.12.09.0"}}}
```
Comment (abhay) [inline rebar.config]:
```suggestion
    {blockchain, {git, "https://github.com/helium/blockchain-core.git", {tag, "2021.12.14.0"}}}
```

Comment (abhay) [reply rebar.config]: need to also bump the lock file?

File: config/sys.config (hunk 2 above)
Comment (jeffgrunewald) [reply config/sys.config]: thanks!

File: config/sys.config (hunk 1 above)
Comment (jeffgrunewald) [reply config/sys.config:37]: @abhay got this for me

File: rebar.config (hunk 4 above)
Comment (jeffgrunewald) [reply rebar.config]: done

**Review by madninja — APPROVED** [review]

**Pull request merged**